spec/factories/completed_lessons.rb

```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :completed_lesson do
+    lesson_id {1}
+    user_id {1}
```

Comment:
To build valid associations within a factor you can do
```ruby
  factory :completed_lesson do
    lesson
    user
```

FactoryBot will search for `user` and `leasson` factories, build them and associate with `completed_lesson`. This makes tests simpler and more reliable.

Documentation: https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations

Reply:
This was not my own code, so I was hesitant to change it and tried to work around it.  However, based on your notes, I did change it to simplify and make the tests more reliable. I got my own tests work.  However, it made two existing tests fail.  I will look into this further to see what can be done for the existing tests.